Documentation/network/bgp-control-plane.rst

```diff
@@ -337,69 +337,87 @@ a BGP control plane.
 Agent-Side Architecture
 ~~~~~~~~~~~~~~~~~~~~~~~
 
-Controller
-^^^^^^^^^^
+At a high level, the ``Agent-Side Control Plane`` is divided into following 
+sub-modules:
+
+- Agent
+- BGP Manager  
+- BGP Router 
+
+
+Agent
+^^^^^
 
-The ``Agent-Side Control Plane`` implements a controller located in
-``pkg/bgpv1/agent/controller.go``.
+The ``Agent`` implements a controller located in ``pkg/bgpv1/agent/controller.go``.
 
-The controller listens for ``CiliumBGPPeeringPolicy``, determines if a
-policy applies to its current host and if it does, captures some
-information about Cilium's current state then calls down to the
-implemented ``BGPRouterManager``.
+The controller listens for ``CiliumBGPPeeringPolicy`` changes and 
+determines if the policy applies to its current host. 
+It will then capture some information about Cilium's current state 
+and pass down the desired state to ``BGP Manager``.
 
-BGPRouterManager
-^^^^^^^^^^^^^^^^
+BGP Manager
+^^^^^^^^^^^
 
-The ``BGPRouterManager`` is an interface used to define a declarative
-API between the ``Controller`` and instantiated BGP routers.
+The ``BGP Manager`` implements the interface ``BGPRouterManager``, which 
+defines a declarative API between the ``Controller`` and instances of 
+BGP routers.
 
 The interface defines a single declarative method whose argument is the
 desired ``CiliumBGPPeeringPolicy`` (among a few others).
 
-The ``BGPRouterManager`` is in charge of pushing the
-``BGP Control Plane`` to the desired ``CiliumBGPPeeringPolicy`` or
-returning an error if it is not possible.
+The ``BGP Manager`` is in charge of pushing the ``BGP Control Plane`` 
+to the desired ``CiliumBGPPeeringPolicy`` or returning an error if it 
+is not possible.
 
-GoBGP Implementation
-''''''''''''''''''''
+Implementation Details
+''''''''''''''''''''''
 
-The first implementation of ``BGPRouterManager`` utilizes the ``gobgp``
-package. You can find this implementation in ``pkg/bgpv1/gobgp``.
-
-This implementation will:
+``BGP Manager`` implementation will take desired ``CiliumBGPPeeringPolicy`` 
+and translate into imperative router API calls :
 
 -  evaluate the desired ``CiliumBGPPeeringPolicy``
 -  create/remove the desired BGP routers
 -  advertise/withdraw the desired BGP routes
 -  enable/disable any BGP server specific features
 -  inform the caller if the policy cannot be applied
 
-The GoBGP implementation is capable of evaluating each ``CiliumBGPVirtualRouter`` in isolation. This
-means when applying a ``CiliumBGPPeeringPolicy`` the GoBGP ``BGPRouterManager`` will attempt to
-create each ``CiliumBGPVirtualRouter``. If a particular ``CiliumBGPVirtualRouter`` fails to
-instantiate the error is logged and the ``BGPRouterManager`` will continue to the next
-``CiliumBGPVirtualRouter``, utilizing the aforementioned logic.
+The ``BGP Manager`` evaluates each ``CiliumBGPVirtualRouter`` in isolation. 
```

Comment:
Why spaces in `BGP Manager` and `BGP Router`? It does make it slightly easier to read, but not longer corresponds to the real object names in the code, does it?

Reply:
In the code, it is just Manager and Router. I can drop BGP prefix if it makes it easier to read and reflects actual usage in code. Any thoughts ?

Reply:
OK. Or just take the “BGP” part out of the preformatted text in that case, so it's easy to spot the part that corresponds to existing symbols:

```rst
The BGP ``Manager`` ...
```

Reply:
I dropped BGP as a prefix.